lib/src/ruby/fact.cc

```diff
@@ -120,6 +120,21 @@ namespace facter { namespace ruby {
                 // Set the value to what was resolved
                 _value = value;
                 _weight = weight;
+
+                if (! ruby.is_nil(_value)) {
+                  return 0;
+                }
+
+                // None of our resolvers were able to resolve the value.
+                // Thus, we attempt to use the "built-in" fact's value.
+                auto builtin_value = facts[ruby.to_string(_name)];
```

Comment:
Does it make more sense to do this at https://github.com/puppetlabs/facter/blob/b1960ad78d78616cf71a44164e2169801b15db27/lib/src/ruby/module.cc#L360 ?